Backend/Controllers/ProductController.cs

```diff
@@ -0,0 +1,250 @@
+﻿using Data;
+using Data.Models.ProductTables;
+using Backend.Services;
+using Lib.EntityFrameworkCore;
+using Microsoft.AspNetCore.Mvc;
+using Lib.Storage;
+using Lib.Storage.Services;
+using System.Drawing;
+using System.Drawing.Imaging;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.AspNetCore.Authorization;
+using Lib;
+
+namespace Backend.Controllers
+{
+    [Route("v1/products")]
+    [ApiController]
+    [Authorize]
+    public class ProductController : ControllerBase
+    {
+        private readonly Db db;
+
+        private readonly ImageService imageService;
+
+        private readonly IStorage storage;
+
+        private readonly Jwt jwt;
+
+
+        public ProductController(Db context, ImageService imageService, Storj storage, Jwt jwt)
+        {
+            db = context;
+            this.imageService = imageService;
+            this.storage = storage;
+            this.jwt = jwt;
+        }
+
+
+        [HttpPost]
+        public async Task<IActionResult> CreateProduct(Product newProduct)
```

Comment:
Custom class for request body. You don't accept whole product! 
public record CreateProductInput(string Name, ...) <- only values required to create new one.